src/main/java/acme/entities/sponsorships/Sponsorship.java

```diff
@@ -0,0 +1,63 @@
+
+package acme.entities.sponsorships;
+
+import java.time.Duration;
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Past;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Positive;
+
+import org.hibernate.validator.constraints.URL;
+import org.hibernate.validator.constraints.time.DurationMin;
+
+import acme.client.data.AbstractEntity;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+public class Sponsorship extends AbstractEntity {
+
+	//Serialisation identifier -----------------------------------------------------------------
+
+	private static final long	serialVersionUID	= 1L;
+
+	//Attributes --------------------------------------------------------------------------------
+
+	@NotBlank
+	@Column(unique = true)
+	@Pattern(regexp = "[A-Z]{1,3}-[0-9]{3}")
+	private String				code;
+
+	@Past
+	@NotNull
+	@Temporal(TemporalType.TIMESTAMP)
+	private Date				moment;
+
+	@NotNull
+	@DurationMin(days = 30)
+	private Duration			duration;
+
+	@NotNull
+	@Positive
+	private Integer				amount;
+
+	@NotNull
+	private SponsorshipType		type;
+
+	@Email
+	private String				contactEmail;
+
+	@URL
```

Comment:
Add length constraints (between 0 and 255 according to on your tutorials, in url and email fields)
[https://ev.us.es/webapps/discussionboard/do/message?action=list_messages&course_id=_85092_1&nav=discussion_board_entry&conf_id=_405265_1&forum_id=_234042_1&message_id=_405561_1](url)